autogen/coding/local_commandline_code_executor.py

```diff
@@ -242,7 +262,22 @@ def _execute_code_dont_check_setup(self, code_blocks: List[CodeBlock]) -> Comman
                 f.write(code)
             file_names.append(written_file)
 
-            program = sys.executable if lang.startswith("python") else _cmd(lang)
+            # Check if the code will be executed.
+            program = None
+            if lang.startswith("python"):
+                program = sys.executable
+            else:
+                try:
+                    program = _cmd(lang)
```

Comment:
there can implement like this:
1. put this using function _cmd to instance method, so user can overwrite it
2. _cmd can return  None,means there is no need to execute

```
program = self.get_lang_cmd(lang)
if program is None:
    logs_all += f"Code saved to {str(written_file)}\n"
    exitcode = 0
    continue
```